system/security/plugins/jwtSecurity/jwtCache.cpp

```diff
@@ -225,36 +225,62 @@ bool JWTUserInfo::hasFileScopePerms() const
 // Private methods
 //-----------------------------------------------------------------------------
 
-// regex patterns for converting a filename glob pattern
-// to an example string
-static const std::regex _globRegex1("\\[![0-9].+?\\]\\*?");                 // foo[!1] -> fooa, foo[!1]* -> fooa
-static const std::regex _globRegex2("\\[([0-9]|![A-Za-z]).+?\\]\\*?");      // foo[1] -> foo1, foo[1]* -> foo1, foo[!m] -> foo1, foo[!m]* -> foo1
-static const std::regex _globRegex3("\\[[A-Za-z].+?\\]\\*?");               // foo[m] -> fooa, foo[m]* -> fooa
-static const std::regex _globRegex4("[*?]");                                // foo* -> fooa, foo? -> fooa
+std::string JWTUserInfo::_convertPathname(std::string logicalPath) const
+{
+    if (!logicalPath.empty())
+    {
+        char lastChar = logicalPath.back();
 
-std::string JWTUserInfo::_globToExample(const std::string& pattern) const
+        if (lastChar == ':')
+        {
+            logicalPath += "*";
+        }
+        else if (lastChar != '*' && lastChar != '?')
+        {
+            logicalPath += "::*";
+        }
+    }
+
+    return std::regex_replace(logicalPath, std::regex("\\*"), "[^:]*");
+}
+
+unsigned int JWTUserInfo::_countDelimitersInPath(const std::string& path) const
 {
-    std::string s1 = std::regex_replace(pattern, _globRegex1, "a");
-    std::string s2 = std::regex_replace(s1, _globRegex2, "1");
-    std::string s3 = std::regex_replace(s2, _globRegex3, "a");
-    std::string s4 = std::regex_replace(s3, _globRegex4, "a");
+    unsigned int    count = 0;
+    std::string     delim("::");
+
+    for (size_t offset = path.find(delim); offset != std::string::npos; offset = path.find(delim, offset + delim.length()))
+    {
+        ++count;
+    }
 
-    return s4;
+    return count;
 }
 
-std::string JWTUserInfo::_convertPathname(const std::string& logicalPath) const
+void JWTUserInfo::_clampPathByDelimCount(const std::string& path, unsigned int delimCount, std::string& clampedPath) const
 {
-    std::string newPath(std::regex_replace(logicalPath, std::regex("::"), "/"));
+    unsigned int    count = 0;
+    std::string     delim("::");
 
-    if (!newPath.empty())
+    clampedPath.clear();
+
+    for (size_t offset = path.find(delim); offset != std::string::npos; offset = path.find(delim, offset + delim.length()))
     {
-        if (newPath.back() == '/')
-            newPath += "*";
-        else if (newPath.back() != '*')
-            newPath += "/*";
+        if (count < delimCount)
+        {
+            ++count;
+        }
+        else
+        {
+            clampedPath = path.substr(0, offset);
+            break;
```

Comment:
If this is changed to `return;`, clearing `clampedPath` on 265 and checking its length on 280 would not be necessary.

Reply:
Agreed and changed.  In addition, the empty() test become superfluous, so I have removed it.

Reply:
Clearing `clampedPath` on line 265 is still happening.

Reply:
Removed defensive line.